composer/trainer/trainer.py

```diff
@@ -1020,18 +1038,27 @@ def _determine_autoresume_load_path(
         if save_latest_filename is None:
             raise ValueError(
                 "save_latest_filename must be specified so autoresume knows where to load checkpoints from.")
+        # Check run_name is provided if necessary
         if run_name is None:
-            raise ValueError("run_name must be specified so autoresume knows which run to load from.")
-        latest_filename_formatted = format_name_with_dist(save_latest_filename, run_name)
-        latest_checkpoint_path = os.path.join(save_folder, latest_filename_formatted)
+            for save_arg in [save_folder, save_latest_filename, save_latest_artifact_name]:
+                for _, field_name, _, _ in string.Formatter().parse(save_arg):
+                    if field_name == "run_name":
+                        raise ValueError("run_name must be specified so autoresume knows which run to load from.")
```

Comment:
```suggestion
                        raise ValueError("Since the `{save_arg}` contains the {{run_name}} variable, the `run_name` argument must be specified to determine the checkpoint path and load the checkpoint.")
```